src/pages/sign-up/sign-up-page.tsx

```diff
@@ -2,47 +2,151 @@ import React, { useState } from 'react';
-        inputType="password"
-        variant="dark"
-        value={confirmPassword}
-        onChange={setConfirmPassword}
-      />
+      <div className="field-container">
+        <TextBox label="username" variant="dark" value={userName} onChange={setUserName} />
+        {_getLabelError('', (formError) => formError.username)}
+      </div>
+
+      <div className="field-container">
+        <TextBox label="email" variant="dark" value={email} onChange={setEmail} />
+        {_getLabelError('', (formError) => formError.email)}
+      </div>
+
+      <div className="field-container">
+        <div className="password-fields">
+          <TextBox
+            label="password"
+            variant="dark"
+            inputType="password"
+            value={password}
+            onChange={setPassword}
+          />
+          <TextBox
+            label="confirm password"
+            inputType="password"
+            variant="dark"
+            value={confirmPassword}
+            onChange={setConfirmPassword}
+          />
+        </div>
+        {_getLabelError(
+          'Use 6 or more characters with at least one of each: upper & lower letters, numbers, symbols.',
+          (formError) => formError.password
+        )}
+      </div>
     </RegistrationPanel>
   );
 
-  function handleSubmitClick() {
-    console.log(userName, password);
-  }
-
   function handleSwapPanelClick() {
     navigate(paths.signIn);
   }
+
+  async function handleSubmitClick() {
+    // simple client side verifications
+    {
+      const newFormError: FormError = {};
+      if (userName === '') {
+        newFormError.username = 'Username should not be empty.';
+      }
+      if (email === '') {
+        newFormError.email = 'Email should not be empty.';
+      }
+      if (!email.includes('@')) {
+        newFormError.email = `This doesn't appear to be a valid email.`;
+      }
+      if (password != confirmPassword) {
+        newFormError.password = `Those passwords didn't match. Try again.`;
+      }
+      if (password.length < 6) {
+        newFormError.password = `Use 6 characters or more for your password.`;
+      }
+
+      if (!isEmptyObject(newFormError)) {
+        setFormError(newFormError);
+        return;
+      }
+    }
+
+    // let server-side handle complex validations
+    // TODO: we should probably validate password security before sending requests?
+    const data = await userClient.register({
+      username: userName,
+      email,
+      password,
+      phoneNumber: '', // TODO: phone number in UI?
+    });
+
+    if (!data) {
+      throw new Error('An unknown error occurred while submitting the registration form');
+    }
+
+    if (data.statusCode === 200) {
+      const { id } = data.response;
+      console.log('Successfully registered:', id);
+      navigate(paths.signIn);
+      // TODO: redirect to sign up success page with button to login
+      // I don't think we should automatically log users in
```

Comment:
Is there a reason why we don't let them in immediately after creating an account?

Reply:
i think this applies more to apps that have you verify an email + ability to change password

it's usually a good idea to let users try to login but it's different here i guess
no preference really so i'll change it to just log them in

Reply:
oh, I see. I'm down with keeping it as is. Either way